Plugins/Json/include/Acts/Plugins/Json/JsonHierarchicalObjectConverter.hpp

```diff
@@ -0,0 +1,179 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2017-2019 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <Acts/Geometry/TrackingVolume.hpp>
+#include <Acts/Surfaces/Surface.hpp>
+#include "Acts/Geometry/HierarchicalGeometryContainer.hpp"
+#include "Acts/Geometry/TrackingGeometry.hpp"
+#include "Acts/Utilities/Definitions.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <nlohmann/json.hpp>
+
+#include <map>
+
+namespace Acts {
+
+/// Convert Hierarchical Object Container to Json file and vice-versa
+template <typename object_t>
+class JsonHierarchicalObjectConverter {
```

Comment:
Suggestion: `HierarchicalGeometryContainerJsonConverter` is a bit more consistent (although a bit longer).

Reply:
If I look at the root and json converter they always start with json (or root).
But I do think that `JsonHierarchicalGeometryConverter` would be better. I would still drop the `Container` to shorten the name and since what we are converting is not the container but its content.

Reply:
I do not like it, but using `Json...Converter` is consistent with the rest of the code base.

I would still keep the full name `JsonHierarchicalGeometryContainerConverter`. We are not just converting a list of objects but the full container with a stored hierarchy.

Reply:
OK lets go for that ! While we are on the topic of name do you have any preference for the one using tracking geometry ?